modal/file_io.py

```diff
@@ -144,11 +144,12 @@ class _FileIO(Generic[T]):
     _task_id: str = ""
     _file_descriptor: str = ""
     _client: _Client
-    _watch_output_buffer: list[Optional[bytes]] = []
+    _watch_output_buffer: list[Union[Optional[bytes],Exception]] = []
 
     def __init__(self, client: _Client, task_id: str) -> None:
         self._client = client
         self._task_id = task_id
+        self._watch_output_buffer = []
```

Comment:
This was missing, which I believe caused events to be passed around completely at random.

Reply:
@mwaskom WDYT?

Reply:
Yeah it looks like a bug to have default values for the class variables. This would have basically shared a list across all instances of the `FileIO`. I think basically all of the default assignment should move from the class scope into the constructor (although this is the nastiest one since it's mutable and only used mutably afaict).